src/snowflake/snowpark/mock/plan.py

```diff
@@ -314,6 +315,18 @@ def execute_mock_plan(
                 res_df = res_df[
                     ~(res_df.isin(cur_df.values.ravel()).all(axis=1)).values
                 ].drop_duplicates()
+            elif operator == INTERSECT:
+                # Dedup all none rows
+                if res_df.isnull().all(axis=1).where(lambda x: x).count() > 1:
+                    res_df = res_df.drop(index=res_df.isnull().all(axis=1).index[1:])
```

Comment:
Can we have a helper function for this? we have something very similar in `EXCEPT`